src/CMakeLists.txt

```diff
@@ -387,6 +387,73 @@ if(WITH_DPDK)
   list(APPEND ceph_common_deps common_async_dpdk)
 endif()
 
+if(WITH_PMDK)
+  # used in src/librbd/CMakeLists.txt and src/os/CMakeLists.txt
+  if(WITH_PMDK_PKG)
+    # Use installed libpmem-dev and libpmemobj-dev. This will make
+    # libpmem1 and libpmemobj1 build dependencies for Ceph, and install dependencies for
+    # anything that links to them (librbd).
+    find_package(libpmem-dev)
+    list(APPEND EXTRALIBS "pmem")
+    find_package(libpmemobj-dev)
+    list(APPEND EXTRALIBS "pmemobj")
+  else(WITH_PMDK_PKG)
+    # Use locally built and statically linked libpmem and libpmemobj
+
+    include(ExternalProject)
+    ExternalProject_Add(nvml_ext
+      DOWNLOAD_DIR ${CMAKE_BINARY_DIR}/src/
+      GIT_REPOSITORY "https://github.com/ceph/nvml.git"
+      GIT_TAG "dd622819dd4ee97d3920f913c70be"
+      SOURCE_DIR ${CMAKE_BINARY_DIR}/src/nvml
```

Comment:
Just how old is this NVML tag? It might be better to express this in the form of a release tag, or at lest state what NVML release this commit should correspond to.